### DISCUSSION TITLE
Post session plugin activation secret setup

### DISCUSSION THREAD

**Discussion by mensfeld** [jeremyevans/roda#331] [General]
I use Roda in a gem (karafka-web) and I want my own configuration flow to be able to setup the sessions plugin secret. Is there a way?

At the moment I have to:

```ruby
plugin :sessions, key: '_karafka_session', secret: 'a' * 64
```

which runs when I load the code.

I would love to be able to do something like:

```ruby
plugin :sessions, key: '_karafka_session', secret: -> { 'a' * 64 }
```

or any other way where I could alter the secret from a configuration I built after loading Roda app with its code but before starting it.

Is there a way?

**Comment by jeremyevans** [comment]
If it will work for your use case, just load the plugin again after you know what the secret is, before freezing the application:

```ruby
# Initially with bogus secret
plugin :sessions, key: '_karafka_session', secret: 'a' * 64

# later with actual secret, before freezing application
Karafka::Web::Ui::Base.plugin :sessions,  secret: 'b' * 64
```

Like most Roda plugins that accept options, you can load the sessions plugin multiple times and have the new options merged into the previous options.

**Reply by mensfeld** [reply]
Works as expected. Amazing. Thank you
